stdlib/immutablearray.md

```diff
@@ -0,0 +1,1069 @@
+
+### ImmutableArray.**length**
+
+<details disabled>
+<summary tabindex="-1">Added in <code>next</code></summary>
+No other changes yet.
+</details>
+
+```grain
+length : ImmutableArray<a> -> Number
+```
+
+Provides the length of the input array.
+
+Parameters:
+
+|param|type|description|
+|-----|----|-----------|
+|`array`|`ImmutableArray<a>`|The array to inspect|
+
+Returns:
+
+|type|description|
+|----|-----------|
+|`Number`|The number of elements in the array|
+
+Examples:
+
+```grain
+length(fromList([1, 2, 3, 4, 5])) == 5
+```
+
+### ImmutableArray.**getOpt**
+
+<details disabled>
+<summary tabindex="-1">Added in <code>next</code></summary>
+No other changes yet.
+</details>
+
+```grain
+getOpt : (Number, ImmutableArray<a>) -> Option<a>
+```
+
+Attempts to retrieve the element from the array at the specified index.
+A negative index is treated as an offset from the end of the array.
+
+Parameters:
+
+|param|type|description|
+|-----|----|-----------|
+|`index`|`Number`|The index to access|
+|`array`|`ImmutableArray<a>`|The array to access|
+
+Returns:
+
+|type|description|
+|----|-----------|
+|`Option<a>`|`Some(element)` containing the element from the array at the given index or `None` if the index is out of the array bounds|
+
+Examples:
+
+```grain
+getOpt(1, fromList([1, 2, 3])) == Some(2)
+```
+
+```grain
+getOpt(-1, fromList([1, 2, 3])) == Some(3)
+```
+
+```grain
+getOpt(4, fromList([1, 2, 3])) == None
+```
+
+### ImmutableArray.**get**
+
+<details disabled>
+<summary tabindex="-1">Added in <code>next</code></summary>
+No other changes yet.
+</details>
+
+```grain
+get : (Number, ImmutableArray<a>) -> a
+```
+
+Retrieves the element from the array at the specified index.
+A negative index is treated as an offset from the end of the array.
+
+Parameters:
+
+|param|type|description|
+|-----|----|-----------|
+|`index`|`Number`|The index to access|
+|`array`|`ImmutableArray<a>`|The array to access|
+
+Returns:
+
+|type|description|
+|----|-----------|
+|`a`|The element from the array|
+
+Examples:
+
+```grain
+get(1, fromList([1, 2, 3, 4, 5])) == 2
+```
+
+### ImmutableArray.**set**
+
+<details disabled>
+<summary tabindex="-1">Added in <code>next</code></summary>
+No other changes yet.
+</details>
+
+```grain
+set : (Number, a, ImmutableArray<a>) -> ImmutableArray<a>
+```
+
+Creates a new array in which the element at the specified index is set to a
+new value. A negative index is treated as an offset from the end of the array.
+
+Parameters:
+
+|param|type|description|
+|-----|----|-----------|
+|`index`|`Number`|The index to update|
+|`value`|`a`|The value to store|
+|`array`|`ImmutableArray<a>`|The array to update|
+
+Returns:
+
+|type|description|
+|----|-----------|
+|`ImmutableArray<a>`|A new array containing the new element at the given index|
+
+Examples:
+
+```grain
+set(1, 9, fromList([1, 2, 3, 4, 5])) == fromList([1, 9, 3, 4, 5])
+```
+
+### ImmutableArray.**Builder**
```

Comment:
Why did this show up here? Does it need to be moved to the top of the file so it is hidden?

Reply:
It should indeed be moved. Upon moving it up to the top, I am now getting an ambiguity warning about `Builder` and `ImmutableArray` sharing a `tail` field; appears to be #775 . I'll just change `tail` from `Builder` to `btail` for now.